app/src/main/java/org/stepic/droid/ui/adapters/SearchQueriesAdapter.kt

```diff
@@ -0,0 +1,98 @@
+package org.stepic.droid.ui.adapters
+
+import android.support.v4.content.ContextCompat
+import android.support.v7.widget.RecyclerView
+import android.support.v7.widget.SearchView
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.style.ForegroundColorSpan
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.search_query_item.view.*
+import org.stepic.droid.R
+import org.stepic.droid.analytic.Analytic
+import org.stepic.droid.base.App
+import org.stepic.droid.model.SearchQuery
+import org.stepic.droid.model.SearchQuerySource
+import javax.inject.Inject
+
+
+class SearchQueriesAdapter : RecyclerView.Adapter<SearchQueriesAdapter.SearchQueryViewHolder>() {
+    @Inject
+    lateinit var analytic: Analytic
+
+    init {
+        App.component().inject(this)
+    }
+
+    private var items: List<Pair<Spannable, SearchQuerySource>> = emptyList()
+
+    var rawDBItems: List<SearchQuery> = emptyList()
+        set(value) {
+            field = value
+            filterItems()
+        }
+
+    var rawAPIItems: List<SearchQuery> = emptyList()
+        set(value) {
+            field = value
+            filterItems()
+        }
+
+    var constraint: String = ""
+        set(value) {
+            field = value
+            filterItems()
+        }
+
+    var searchView: SearchView? = null
+
+    private lateinit var querySpan: ForegroundColorSpan
+
+    override fun onAttachedToRecyclerView(recyclerView: RecyclerView) {
+        querySpan = ForegroundColorSpan(ContextCompat.getColor(recyclerView.context, R.color.search_view_suggestions_prefix_color))
```

Comment:
it is not correct. Same adapters can be observed by multiple recyclers by design. Let's pass context to constructor and `querySpan` in declaration